### PR TITLE
Feature/roe 1597 trust corporate validation

### PR DESCRIPTION
**Pull request by pshdev0**
### JIRA link

[ROE-1597](https://companieshouse.atlassian.net/browse/ROE-1597)

### Change description

Completed ROE-1597:
1. validation of all fields
2. Unit tests

### Work checklist

- [X] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

[ROE-1597]: https://companieshouse.atlassian.net/browse/ROE-1597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        for (TrustDataDto trustDataDto : trustDataDtoList) {
+            List<TrustCorporateDto> corporates = trustDataDto.getCorporates();
+
+            if (!CollectionUtils.isEmpty(corporates)) {
+                for (TrustCorporateDto trustCorporateDto : corporates) {
+                    validateName(trustCorporateDto.getName(), errors, loggingContext);
+
+                    String type = trustCorporateDto.getType();
+                    if(validateType(type, errors, loggingContext) && BeneficialOwnerType
+                            .findByBeneficialOwnerTypeString(type).equals(BeneficialOwnerType.INTERESTED_PERSON)) {
```
Comment (ch-tstrickland) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java:45]: Could this equals be the other way round perhaps? Less chance of a NullPointerEx if findByBeneficialOwnerTypeString returns null, which probably shouldn't happen, but still...

**Review by ch-tstrickland — COMMENTED** [review]
Aside from my one minor comment, this looks great.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java (hunk 1 above)
Comment (ch-tstrickland) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java:45]: I have had a word with Naz, who has been talking with Paul, and am happy to disregard this comment.

**Review by nawazish587 — APPROVED** [review]
Tom reviewed and okay with the changes. I am approving PR on his behalf.

**Pull request merged**